core/src/main/scala/org/scalafmt/rewrite/ReplaceForParensWithBraces.scala

```diff
@@ -0,0 +1,92 @@
+package org.scalafmt.rewrite
+
+import org.scalafmt.util.{TokenOps, Whitespace}
+
+import scala.annotation.tailrec
+import scala.meta.tokens.Tokens
+import scala.meta.{Tree, _}
+
+/**
+  * Replaces multi generator For / ForYield Expression parens and semi-colons
+  * with braces and new-lines.
+  *
+  * For example,
+  *
+  *   for(a <- as; b <- bs if b > 2) yield (a, b)
+  *
+  * becomes,
+  *
+  *   for {
+  *     a <- as
+  *     b <- bs if b > 2
+  *   } yield (a, b)
+  *
+  */
+object ReplaceForParensWithBraces extends Rewrite {
+
+  def findForParens(forTokens: Tokens, ctx: RewriteCtx): Option[(Token, Token)] = {
+    import ctx.tokenTraverser._
+
+    @tailrec
+    def nextValidLeftParen(token: Token):Option[Token] = {
```

Comment:
That's a great suggestion. The only change I'd like to make is on the return type. I think `Option[Token]` would be more idiomatic.

``` scala
def find(start: Token)(f: Token => Boolean): Option[Token]
def reverseFind(start: Token)(f: Token => Boolean): Option[Token]
```